### PR TITLE
Use Crystal 1.1.1

### PR DESCRIPTION
**Pull request by dentarg**
The Crystal install script has deprecated the `--crystal` flag, and it doesn't
support patch versions anymore, just major.minor.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
         run: |
           wget https://crystal-lang.org/install.sh
           chmod +x install.sh
-          sudo ./install.sh --crystal=1.0.0
+          sudo ./install.sh --version=1.1
```
Comment (carlhoerberg) [inline .github/workflows/ci.yml:31]: Or should we start using the container here too?

Comment (dentarg) [reply .github/workflows/ci.yml:31]: I think we are not using the container here because it is easier (and more performant?) to install RabbitMQ in Ubuntu

Comment (dentarg) [reply .github/workflows/ci.yml:31]: ...with that said, if we want to use container for Crystal, we could switch to a service container for RabbitMQ, but it is a bit more Yaml needed then, example: https://github.com/dentarg/actions-test/blob/a721484c6b99d271d20779159bff8b3c2720fbf9/.github/workflows/rabbitmq.yml#L14-L27

**Comment by dentarg** [comment]
Done in https://github.com/cloudamqp/amqproxy/commit/3ddacf0212e6d9ccac200b4d436ed15fe3a637b3#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f

**Comment by carlhoerberg** [comment]
Yes, but failing due to bad specs etc. Is localhost not the server to connect to when connecting to a "service"?

**Comment by dentarg** [comment]
> Yes, but failing due to bad specs etc. Is localhost not the server to connect to when connecting to a "service"?

No, not when we run in a container. We need to change the specs so they read some ENVs we can set from the workflow, see this example with postgres: https://github.com/dentarg/actions-test/blob/a721484c6b99d271d20779159bff8b3c2720fbf9/.github/workflows/ruby-postgres.yml#L41-L46

Here are the actions docs on this: https://docs.github.com/en/actions/using-containerized-services/about-service-containers

**Pull request closed without merging**